File: Scripts/results/generated_code_results/Flutter/bak_code/Social Networking/SocialApp/SearchResultScreen.dart
import 'package:flutter/material.dart';
class SearchResultScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
backgroundColor: Colors.white,
elevation: 0,
leading: IconButton(
icon: Icon(Icons.arrow_back, color: Colors.black),
onPressed: () => Navigator.pop(context),
),
title: TextField(
decoration: InputDecoration(
hintText: 'Gracia',
prefixIcon: Icon(Icons.search),
border: InputBorder.none,
),
),
),
body: SingleChildScrollView(
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Padding(
padding: EdgeInsets.all(16.0),
child: Text(
'Related Result',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
),
_buildProfileCard(
name: 'Klee Gracia',
age: 24,
interests: ['Running', 'Outdoor', 'Anime'],
bio: "Hi there! I'm Gracia ✨, a fun-loving adventurer looking for great connections.",
isPotentialMatch: true,
context: context,
),
Padding(
padding: EdgeInsets.all(16.0),
child: Text(
'More About Gracia',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
),
_buildRelatedProfileGrid(context),
],
),
),
);
}
Widget _buildProfileCard({
required String name,
required int age,
required List<String> interests,
required String bio,
required bool isPotentialMatch,
required BuildContext context,
}) {
return GestureDetector(
onTap: () => Navigator.pushNamed(context, 'Profile'),
child: Card(
margin: EdgeInsets.symmetric(horizontal: 16, vertical: 8),
shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16)),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Stack(
children: [
ClipRRect(
borderRadius: BorderRadius.vertical(top: Radius.circular(16)),
child: Image.asset('assets/images/template.png', fit: BoxFit.cover, height: 200, width: double.infinity),
),
if (isPotentialMatch)
Positioned(
top: 16,
left: 16,
child: Container(
padding: EdgeInsets.symmetric(horizontal: 8, vertical: 4),
decoration: BoxDecoration(
color: Colors.black.withOpacity(0.6),
borderRadius: BorderRadius.circular(16),
),
child: Row(
children: [
Icon(Icons.bolt, color: Colors.yellow, size: 16),
SizedBox(width: 4),
Text('Potential Match!', style: TextStyle(color: Colors.white, fontSize: 12)),
],
),
),
),
],
),
Padding(
padding: EdgeInsets.all(16),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Row(
children: [
Text('$name, $age', style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold)),
SizedBox(width: 8),
Icon(Icons.verified, color: Colors.blue, size: 20),
],
),
SizedBox(height: 8),
Wrap(
spacing: 8,
children: interests.map((interest) => Chip(label: Text(interest))).toList(),
),
SizedBox(height: 8),
Text(bio, style: TextStyle(fontSize: 14)),
],
),
),
],
),
),
);
}
Widget _buildRelatedProfileGrid(BuildContext context) {
final relatedProfiles = [
{'name': 'Mika Gracia', 'age': 18, 'location': 'Bali, Indonesia', 'matchPercentage': 98},
{'name': 'Gracy Logan', 'age': 28, 'location': 'Jakarta, Indonesia', 'matchPercentage': 98},
{'name': 'R.O Gracia', 'age': 17, 'location': 'Bekasi, Indonesia', 'matchPercentage': 93},
{'name': 'Audrey Gracia', 'age': 17, 'location': 'Bali, Indonesia', 'matchPercentage': 93},
];
reasonml
复制
return GridView.builder(
  shrinkWrap: true,
  physics: NeverScrollableScrollPhysics(),
  padding: EdgeInsets.symmetric(horizontal: 16),
  gridDelegate: SliverGridDelegateWithFixedCrossAxisCount(
    crossAxisCount: 2,
    childAspectRatio: 0.7,
    crossAxisSpacing: 16,
    mainAxisSpacing: 16,
  ),
  itemCount: relatedProfiles.length,
  itemBuilder: (context, index) {
    final profile = relatedProfiles[index];
    return GestureDetector(
      onTap: () => Navigator.pushNamed(context, 'Profile'),
      child: Container(
        decoration: BoxDecoration(
          borderRadius: BorderRadius.circular(16),
          image: DecorationImage(
            image: AssetImage('assets/images/template.png'),
            fit: BoxFit.cover,
          ),
        ),
        child: Container(
          decoration: BoxDecoration(
            borderRadius: BorderRadius.circular(16),
            gradient: LinearGradient(
              begin: Alignment.topCenter,
              end: Alignment.bottomCenter,
              colors: [Colors.transparent, Colors.black.withOpacity(0.8)],
            ),
          ),
          padding: EdgeInsets.all(12),
          child: Column(
            crossAxisAlignment: CrossAxisAlignment.start,
            mainAxisAlignment: MainAxisAlignment.end,
            children: [
              Text(
                '${profile['matchPercentage']}% Match',
                style: TextStyle(color: Colors.white, fontWeight: FontWeight.bold),
              ),
              SizedBox(height: 4),
              Text(
                '${profile['name']}, ${profile['age']}',
                style: TextStyle(color: Colors.white, fontSize: 16, fontWeight: FontWeight.bold),
              ),
              SizedBox(height: 4),
              Text(
                '${profile['location']}',
                style: TextStyle(color: Colors.white70, fontSize: 12),
              ),
            ],
          ),
        ),
      ),
    );
  },
);
}
}

File: Scripts/evaluations/rq3_userstudy/user4/TestCases/1/SocialApp/SearchResultScreen.js
import React from 'react';
import { View, Text, Image, TextInput, TouchableOpacity, FlatList, StyleSheet, SafeAreaView, StatusBar } from 'react-native';
import { useNavigation } from '@react-navigation/native';
const SearchResults = () => {
const navigation = useNavigation();
const searchResults = [
{
id: '1',
name: 'Klee Gracia',
age: 24,
bio: "Hi there! I'm Gracia ✨, a fun-loving adventurer looking for great connections.",
interests: ['Running', 'Outdoor', 'Anime'],
potentialMatch: true,
image: require('../assets/snack-icon.png'),
},
];
const relatedProfiles = [
{ id: '2', name: 'Mika Gracia', age: 18, location: 'Bali, Indonesia', matchPercentage: 98, image: require('../assets/snack-icon.png') },
{ id: '3', name: 'Gracy Logan', age: 28, location: 'Jakarta, Indonesia', matchPercentage: 98, image: require('../assets/snack-icon.png') },
{ id: '4', name: 'R.O Gracia', age: 17, location: 'Bekasi, Indonesia', matchPercentage: 93, image: require('../assets/snack-icon.png') },
{ id: '5', name: 'Audrey Gracia', age: 17, location: 'Bali, Indonesia', matchPercentage: 93, image: require('../assets/snack-icon.png') },
];
const handleBack = () => {
navigation.goBack();
};
const handleProfilePress = (profile) => {
navigation.navigate('Profile', { profile });
};
const renderSearchResult = ({ item }) => (
<TouchableOpacity style={styles.searchResultCard} onPress={() => handleProfilePress(item)}>
<Image source={item.image} style={styles.profileImage} />
<View style={styles.profileInfo}>
<Text style={styles.name}>{item.name}, {item.age}</Text>
<View style={styles.interestsContainer}>
{item.interests.map((interest, index) => (
<Text key={index} style={styles.interest}>{interest}</Text>
))}
</View>
<Text style={styles.bio} numberOfLines={2}>{item.bio}</Text>
</View>
{item.potentialMatch && (
<View style={styles.potentialMatchBadge}>
<Text style={styles.potentialMatchText}>Potential Match!</Text>
</View>
)}
</TouchableOpacity>
);
const renderRelatedProfile = ({ item }) => (
<TouchableOpacity style={styles.relatedProfileCard} onPress={() => handleProfilePress(item)}>
<Image source={item.image} style={styles.relatedProfileImage} />
<Text style={styles.relatedProfileName}>{item.name}, {item.age}</Text>
<Text style={styles.relatedProfileLocation}>{item.location}</Text>
<Text style={styles.matchPercentage}>{item.matchPercentage}% Match</Text>
</TouchableOpacity>
);
return (
<SafeAreaView style={styles.container}>
<StatusBar barStyle="dark-content" />
<View style={styles.header}>
<TouchableOpacity onPress={handleBack}>
<Image source={require('../assets/snack-icon.png')} style={styles.backIcon} />
</TouchableOpacity>
<View style={styles.searchBar}>
<Image source={require('../assets/snack-icon.png')} style={styles.searchIcon} />
<TextInput style={styles.searchInput} placeholder="Gracia" placeholderTextColor="#888" />
</View>
</View>
  <Text style={styles.sectionTitle}>Related Result</Text>
  <FlatList
    data={searchResults}
    renderItem={renderSearchResult}
    keyExtractor={item => item.id}
    horizontal={false}
  />

  <Text style={styles.sectionTitle}>More About Gracia</Text>
  <FlatList
    data={relatedProfiles}
    renderItem={renderRelatedProfile}
    keyExtractor={item => item.id}
    numColumns={2}
    columnWrapperStyle={styles.relatedProfilesRow}
  />
</SafeAreaView>
);
};
const styles = StyleSheet.create({
container: {
flex: 1,
backgroundColor: '#fff',
},
header: {
flexDirection: 'row',
alignItems: 'center',
padding: 10,
borderBottomWidth: 1,
borderBottomColor: '#e0e0e0',
},
backIcon: {
width: 24,
height: 24,
marginRight: 10,
},
searchBar: {
flex: 1,
flexDirection: 'row',
alignItems: 'center',
backgroundColor: '#f0f0f0',
borderRadius: 20,
paddingHorizontal: 10,
},
searchIcon: {
width: 20,
height: 20,
marginRight: 10,
},
searchInput: {
flex: 1,
height: 40,
},
sectionTitle: {
fontSize: 18,
fontWeight: 'bold',
margin: 10,
},
searchResultCard: {
flexDirection: 'row',
padding: 10,
borderBottomWidth: 1,
borderBottomColor: '#e0e0e0',
},
profileImage: {
width: 80,
height: 80,
borderRadius: 40,
marginRight: 10,
},
profileInfo: {
flex: 1,
},
name: {
fontSize: 16,
fontWeight: 'bold',
},
interestsContainer: {
flexDirection: 'row',
marginTop: 5,
},
interest: {
backgroundColor: '#e0e0e0',
paddingHorizontal: 8,
paddingVertical: 4,
borderRadius: 10,
marginRight: 5,
fontSize: 12,
},
bio: {
marginTop: 5,
color: '#666',
},
potentialMatchBadge: {
position: 'absolute',
top: 10,
right: 10,
backgroundColor: '#FFD700',
paddingHorizontal: 8,
paddingVertical: 4,
borderRadius: 10,
},
potentialMatchText: {
fontSize: 12,
fontWeight: 'bold',
},
relatedProfilesRow: {
justifyContent: 'space-between',
paddingHorizontal: 10,
},
relatedProfileCard: {
width: '48%',
marginBottom: 10,
borderRadius: 10,
overflow: 'hidden',
backgroundColor: '#f0f0f0',
},
relatedProfileImage: {
width: '100%',
height: 150,
resizeMode: 'cover',
},
relatedProfileName: {
fontSize: 14,
fontWeight: 'bold',
margin: 5,
},
relatedProfileLocation: {
fontSize: 12,
color: '#666',
marginHorizontal: 5,
},
matchPercentage: {
fontSize: 12,
color: '#4CAF50',
fontWeight: 'bold',
margin: 5,
},
});
export default SearchResults;